dags/probe_scraper.py

```diff
@@ -265,17 +260,40 @@
                 "AWS_ACCESS_KEY_ID": aws_access_key,
                 "AWS_SECRET_ACCESS_KEY": aws_secret_key
             },
-            # wait for upstream, ignore upstream failures, and don't run if day_check
-            # status is "skipped", i.e. still only run on specific days
-            trigger_rule="none_skipped",
             dag=dag,
             **airflow_gke_prod_kwargs,
         )
-        day_check >> check_task
-        probe_scraper_glean_repositories >> check_task
-        for glean_task in probe_scraper_glean:
-            glean_task >> check_task
-        probe_scraper_checks.append(check_task)
+        for check_name in ("check-expiry", "check-fog-expiry")
+    ]
+    dummy_branch = DummyOperator(
+        task_id=f"dummy_branch",
+        dag=dag,
+    )
+
+
+    class CheckBranchOperator(BaseBranchOperator):
+        def choose_branch(self, context):
+            """
+            Run an extra branch on the first day of the month
+            """
+            weekday = context["execution_date"].isoweekday()
+            if weekday == WeekDay.MONDAY:
+                return ["probe_scraper_check_expiry"]
+            elif weekday == WeekDay.WEDNESDAY:
+                return ["probe_scraper_check_fog_expiry"]
+            else:
+                return ["dummy_branch"]
+
+
+    check_branch = CheckBranchOperator(
+        task_id="probe_scraper_check_branch",
+        # wait for upstream, but ignore upstream failures
+        trigger_rule="all_done",
+        dag=dag,
+    )
+    check_branch >> [*probe_scraper_checks, dummy_branch]
+    check_branch << probe_scraper_glean
+    check_branch << probe_scraper_glean_repositories
```

Comment:
What do you think about sticking with `>>` operators? There's already a lot of new context in this change and we may be able to avoid some of it

Reply:
as i understand it, you cannot do `[task, ...] >> task`, it has to be `task << [task, ...]` or `task.set_upstream([task, ...]`, and `probe_scraper_glean` is a list

Reply:
In retrospect, that makes total sense. You could iterate through the list but that may be ugly. 👍 to leave as-is as well.

Reply:
turns out i am wrong
```suggestion
    probe_scraper_glean >> check_branch
    probe_scraper_glean_repositories >> check_branch
```